A71-xds-fallback.md

```diff
@@ -0,0 +1,233 @@
+the secondary xDS server is better than nothing but is less desirable than
+the data in the primary server. E.g. a fallback environment that uses
+an outdated static configuration. This means that while we definitely want
+to use the fallback data in preference to the client simply failing, we will
+always prefer to use the data from the primary server over the data from
+the fallback server. In principle, this means that whenever we have a resource
+already cached from the primary server, we want to use that instead of falling
+back.
+
+We have no guarantee that a combination of resources from different xDS servers
+form a valid cohesive configuration, so we cannot make this determination on
+a per-resource basis. We need any given gRPC channel or server listener to only
+use the resources from the single server.
+
+### Related Proposals: 
+* [A27: xDS-Based Global Load Balancing][A27]
+* [A36: xDS-Enabled Servers][A36]
+* [A40: xDS Configuration Dump via Client Status Discovery Service in gRPC][A40]
+* [A47: xDS Federation][A47]
+* [A57: XdsClient Failure Mode Behavior][A57]
+
+## Proposal
+
+Implement the following changes
+
+1. Change global XdsClient from a single instance to a map of instances keyed
+by channel target.
+1. Update CSDS to aggregate configs from multiple XdsClient instances.
+1. Change xDS bootstrap code to return the full list of servers for each
+xDS authority from the bootstrap config.
+1. Change XdsClient to support fallback within each xDS authority.
+
+### Change global XdsClient from a single instance to per-target instances
+
+Current gRPC implementation is using a single XdsClient for all data plane
+targets. For fallback, if any one channel triggers fallback, then all channels
+regardless of authority would switch to using the data from the fallback xDS
+control plane which is not desirable.
+
+gRPC servers using the xDS configuration will share the same XdsClient instance
+keyed with a dedicated well-known key value.
+
+Changes XdsClient to per-data plane target would enable gRPC to switch to
+fallback configuration only for channels that need xDS resources that have not
+been downloaded yet.
+
+This change would change the channels to get XdsClient for their data plane
+target.
+
+### Update CSDS to aggregate configs from multiple XdsClient instances.
+
+The CSDS RPC service will be changed to get data from all XdsClient instances.
+A new string field named `channel_target` will be added to the CSDS
+`ClientConfig` message to indicate the channel target the data is associated
+with.
+
+Adding this field will not require updating the CSDS client as its value will
+be the same for most resources in most cases.
+
+### Change xDS Bootstrap to handle multiple xDS configuration servers
+
+Current gRPC xDS implementations only use the first xDS configuration server
+listed in the bootstrap JSON document. Fallback implementation requires changes
+to use all servers listed in order from highest to lowest priority.
```

Comment:
Would be good to specify that the order establishes the priority with the first being the highest priority.